src/eureka/S1_detector_processing/remove390.py

```diff
@@ -0,0 +1,448 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import os
+from tqdm import tqdm
+import multiprocessing as mp
+
+from astropy.stats import sigma_clip
+from astropy.convolution import Box1DKernel, convolve
+
+from scipy.signal import lombscargle
+from scipy.optimize import minimize
+
+from ..lib import plots
+
+
+def computePixelTimes(integ, grp, SUBSIZE2, SUBSIZE1, NGROUPS, TSAMPLE,
+                      TGROUP):
+    """Compute the time stamps for every pixel in the group.
+
+    Based on code sent to TJB by Michael E Ressler.
+
+    Parameters
+    ----------
+    integ : int
+        The current integration number.
+    grp : int
+        The current group number.
+    SUBSIZE2 : int
+        The subarray ysize.
+    SUBSIZE1 : int
+        The subarray xsize.
+    NGROUPS : int
+        The number of groups per integration.
+    TSAMPLE : float
+        The sample time.
+    TGROUP : float
+        The group time.
+
+    Returns
+    -------
+    pixel_times : np.array
+        The time stamp for every pixel in the group.
+    """
+    # SLITLESSPRISM Subarray
+    nrows = SUBSIZE2  # px
+    ncols = SUBSIZE1  # px
+    ngrp = NGROUPS
+    
+    nreadouts = 4  # There are four readouts that are all simultaneous
+    dt_fourPix = TSAMPLE*1e-6  # s
+    dt_rowSep = dt_fourPix*11  # s (Not sure why it is 11 or if it is always)
+    dt_oneRow = (ncols/nreadouts-1)*dt_fourPix + dt_rowSep  # s
+    dt_grp = TGROUP  # s
+    dt_int = dt_grp*(ngrp+1)  # s
+    
+    # Time steps within one row
+    dt_cols = np.repeat(np.arange(ncols/nreadouts, dtype=float)*dt_fourPix,
+                        nreadouts)
+    
+    # Time steps between rows
+    dt_rows = np.arange(nrows, dtype=float)*dt_oneRow
+    
+    # Reshape and add together to get 2D time array
+    dt_rows = dt_rows.reshape((-1, 1))
+    dt_cols = dt_cols.reshape((1, -1))
+    pixel_times = dt_cols+dt_rows+(dt_grp*grp)+(dt_int*integ)
+    
+    return pixel_times
+
+
+def smoothData(tempdata, boundary='extend', box_width=100):
```

Comment:
Would it make sense to move this function to smooth.py? It might be useful for other routines.

Alternatively, should we use an existing smoothing function?